### PR TITLE
[SAF-20] add manually time controls to charts

### PR DESCRIPTION
**Pull request by luchopcerra**

### CODE REVIEW

**Comment by luchopcerra** [comment]
https://github.com/user-attachments/assets/4adfb9fb-fc73-4fdc-8273-4f38977044f1

File: apps/frontend/src/app/5_pages/AaveReserveOverviewPage/components/BorrowDetailsGraph/BorrowDetailsGraph.tsx [hunk 1]
```diff
@@ -35,12 +35,12 @@ export const BorrowDetailsGraph: FC<BorrowDetailsGraphProps> = ({
     return normalizeBorrowStats(reserve);
   }, [reserve]);
 
-  const [timeRange] = useState<ReserveRateTimeRange>(
+  const [timeRange, setTimeRange] = useState<'1m' | '6m' | '1y'>(
     ESupportedTimeRanges.OneMonth,
   );
```
Comment (matzapata) [inline apps/frontend/src/app/5_pages/AaveReserveOverviewPage/components/BorrowDetailsGraph/BorrowDetailsGraph.tsx]: I think problem is `ReserveRateTimeRange` should be a enum like:

```
enum ReserveRateTimeRange = {
OneMonth = "1m",
...
}
```

File: apps/frontend/src/app/5_pages/AaveReserveOverviewPage/components/BorrowDetailsGraph/components/Chart/Chart.tsx [hunk 2]
```diff
@@ -13,9 +14,10 @@ import { htmlLegendPlugin } from './Chart.utils';
 
 type ChartProps = {
   input: ChartData;
+  onTimeRangeChange: (range: '1m' | '6m' | '1y') => void;
```
Comment (matzapata) [inline apps/frontend/src/app/5_pages/AaveReserveOverviewPage/components/BorrowDetailsGraph/components/Chart/Chart.tsx]: same here, you can use enum, and everywhere else

File: apps/frontend/src/app/5_pages/AaveReserveOverviewPage/components/TimeRangeButtons/TimeRangeButtons.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  };
+
+  return (
+    <div className="flex space-x-2 justify-end mb-4">
+      {' '}
+      <button
+        className={`py-1.5 px-4 rounded-md text-sm font-medium ${
+          activeRange === '1m'
+            ? 'bg-[#484D59] text-white'
+            : 'bg-[#2C303B] text-gray-400'
```
Comment (matzapata) [inline apps/frontend/src/app/5_pages/AaveReserveOverviewPage/components/TimeRangeButtons/TimeRangeButtons.tsx]: what are these hex colors? can't we use tailwind variables?

**Review by matzapata — COMMENTED** [review]
Overall looks good, just this comment about the enum that I think affects all the files

File: apps/frontend/src/app/5_pages/AaveReserveOverviewPage/components/TimeRangeButtons/TimeRangeButtons.tsx (hunk 3 above)
Comment (luchopcerra) [reply apps/frontend/src/app/5_pages/AaveReserveOverviewPage/components/TimeRangeButtons/TimeRangeButtons.tsx]: of course; in fact they are just forget to import and use

**Comment by luchopcerra** [comment]
> Overall looks good, just this comment about the enum that I think affects all the files

i've addressed both. the enum from hook and color variables

File: apps/frontend/src/app/5_pages/AaveReserveOverviewPage/components/TimeRangeButtons/TimeRangeButtons.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  ReserveRateTimeRange,
+} from '../../../../../hooks/aave/useAaveReservesHistory';
+
+interface TimeRangeButtonsProps {
+  onChange: (range: ReserveRateTimeRange) => void;
+}
+
+export const TimeRangeButtons: FC<TimeRangeButtonsProps> = ({ onChange }) => {
+  const [activeRange, setActiveRange] = useState<ReserveRateTimeRange>(
+    ESupportedTimeRanges.OneMonth,
```
Comment (matzapata) [inline apps/frontend/src/app/5_pages/AaveReserveOverviewPage/components/TimeRangeButtons/TimeRangeButtons.tsx:14]: I think this should be `ReserveRateTimeRange.OneMonth` with an enum

Comment (luchopcerra) [reply apps/frontend/src/app/5_pages/AaveReserveOverviewPage/components/TimeRangeButtons/TimeRangeButtons.tsx:14]: `ReserveRateTimeRange` is the type to express 3 posibilities from the enum ESupportedTimeRanges

**Pull request merged**